src/test/java/co/cask/cdc/plugins/integration/sink/CDCBigTablePluginTest.java

```diff
@@ -0,0 +1,387 @@
+package co.cask.cdc.plugins.integration.sink;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.etl.api.batch.SparkSink;
+import co.cask.cdap.etl.mock.spark.streaming.MockSource;
+import co.cask.cdap.etl.proto.v2.ETLPlugin;
+import co.cask.cdap.proto.ProgramRunStatus;
+import co.cask.cdap.test.SparkManager;
+import co.cask.cdc.plugins.common.OperationType;
+import co.cask.cdc.plugins.common.Schemas;
+import co.cask.cdc.plugins.integration.CDCPluginTestBase;
+import co.cask.cdc.plugins.sink.CDCBigTableConfig;
+import co.cask.cdc.plugins.sink.CDCTableUtil;
+import co.cask.hydrator.common.Constants;
+import com.google.bigtable.repackaged.com.google.cloud.ServiceOptions;
+import com.google.bigtable.repackaged.io.grpc.StatusRuntimeException;
+import com.google.cloud.bigtable.hbase.BigtableConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.Get;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.Table;
+import org.assertj.core.api.Assertions;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@RunWith(Enclosed.class)
```

Comment:
I'm not familiar with this, what is the benefit of having inner test classes instead of multiple top level test classes?

Reply:
in general it allows you to encapsulate tests with different preconditions (before/after) and some common logic/variables in one class (easier to describe clear testing conventions - one functionality - one test class).